[masonry] Update to modern track sizing algorithm during accumulating track sizes

stwrt opened the pull request:
#### 7719c4262533e0ce345feda9d30c9b399ec4efbd
<pre>
[masonry] Update to modern track sizing algorithm during accumulating track sizes
<a href="https://bugs.webkit.org/show_bug.cgi?id=268050">https://bugs.webkit.org/show_bug.cgi?id=268050</a>

Reviewed by Sammy Gill.

Update to the new track sizing algorithm. All elements participate that are definite inside the track,
and those that are indefinite.

This patch only handles the algorithm. Another patch will handle placing all indefinite items across
all intrinsic tracks.

Test Case Updates:

masonry-track-sizing-overflow-left/right-side:
These test cases were built off the idea of the old track sizing algorithm where only the first track determined the
column size. Under the new algorithm all items now contribute; therefore the bottom item extends the track an extra 50px.

masonry-gap-001:
Simplified test case to focus on the gap.
Updated the grid-columns for items to ensure that other tracks would not affect the 2nd column.

* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/gap/masonry-gap-001-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/gap/masonry-gap-001-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/gap/masonry-gap-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/masonry-track-sizing-overflow-left-side-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/masonry-track-sizing-overflow-left-side-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/masonry-track-sizing-overflow-right-side-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/masonry-track-sizing-overflow-right-side-ref.html:
* Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp:
(WebCore::GridTrackSizingAlgorithm::accumulateIntrinsicSizesForTrack):

Canonical link: <a href="https://commits.webkit.org/273551@main">https://commits.webkit.org/273551@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4b4ee6d0b26873887c825d467ce5109257835513

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/35735 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/14679 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/37877 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/38465 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32168 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/36953 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/17072 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/11703 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/30940 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/36288 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/12398 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/31784 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/10901 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/10894 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/31942 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39713 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32463 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/32274 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/36850 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11088 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/8983 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/34934 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/12822 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/31604 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8153 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11593 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/11894 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

sammygill commented on LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/gap/masonry-gap-001.html:
Could you put the explanation for the test changes into the commit message (both for the changes to the masonry-gap test and masonry-track-sizing-overflow ones)? I'm assuming it is related to the spec change but it's a little hard to figure out how the changes correspond to it so I think extra information here would help

sammygill reviewed (changes requested):
Let's make sure we have a WPT export for these active also

stwrt committed:
[masonry] Update to modern track sizing algorithm during accumulating track sizes
https://bugs.webkit.org/show_bug.cgi?id=268050

Reviewed by Sammy Gill.

Update to the new track sizing algorithm. All elements participate that are definite inside the track,
and those that are indefinite.

This patch only handles the algorithm. Another patch will handle placing all indefinite items across
all intrinsic tracks.

Test Case Updates:

masonry-track-sizing-overflow-left/right-side:
These test cases were built off the idea of the old track sizing algorithm where only the first track determined the
column size. Under the new algorithm all items now contribute; therefore the bottom item extends the track an extra 50px.

masonry-gap-001:
Simplified test case to focus on the gap.
Updated the grid-columns for items to ensure that other tracks would not affect the 2nd column.

* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/gap/masonry-gap-001-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/gap/masonry-gap-001-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/gap/masonry-gap-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/masonry-track-sizing-overflow-left-side-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/masonry-track-sizing-overflow-left-side-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/masonry-track-sizing-overflow-right-side-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/masonry-track-sizing-overflow-right-side-ref.html:
* Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp:
(WebCore::GridTrackSizingAlgorithm::accumulateIntrinsicSizesForTrack):

Canonical link: https://commits.webkit.org/273551@main
